### PR TITLE
Oxedions/fix/cleanresources

### PR DESCRIPTION
**Pull request by oxedions**
Clean of resources.

Also includes fallback to replace hash behavior, as mainly related to inventory structure.

### CODE REVIEW

File: resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml [hunk 1]
```diff
@@ -3,44 +3,23 @@ nfs_settings:
...
-    clients_groups:
-      - mg_computes
-      - mg_logins
-    take_over_network: interconnect-1
-    export_arguments: rw,no_root_squash,sync
-    mount_arguments: rw,intr,rsize=32768,wsize=32768,nfsvers=4.2,bg
 
   ansible:
     mount: /etc/bluebanquise
     export: /etc/bluebanquise
```
Comment (bouriquet) [inline resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml:9]:
```suggestion
    mount: /etc/bluebanquise                    # Which path clients should mount this NFS
    export: /etc/bluebanquise                   # The server that export this storage space
```

File: resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml [hunk 2]
```diff
@@ -3,44 +3,23 @@ nfs_settings:
...
-    server: odin
+    server: management1
     clients_groups:
```
Comment (bouriquet) [inline resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml:11]:
```suggestion
    server: management1                         # The server that export this storage space
    clients_groups:                             # Can be an equipment group, or a main group (mg), or any other ansible group
```

File: resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml [hunk 3]
```diff
@@ -3,44 +3,23 @@ nfs_settings:
...
-      - equipment_supermicro_management
+      - slave_managements
     take_over_network: ice1-1
     export_arguments: ro,no_root_squash,sync
     mount_arguments: ro,intr,nfsvers=4.2,bg
```
Comment (bouriquet) [inline resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml:15]:
```suggestion
    take_over_network: ice1-1                   # Network used to share this storage space
    export_arguments: ro,no_root_squash,sync    # Arguments for the server (export)
    mount_arguments: ro,intr,nfsvers=4.2,bg     # Arguments for the client (mount)
```

File: resources/examples/simple_cluster/playbooks/login1.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 - name: login1
-  hosts: "{{ target | default('login1') }}"
+  hosts: "login1"
```
Comment (bouriquet) [inline resources/examples/simple_cluster/playbooks/login1.yml:3]: Why single host target instead of login group ?
Documentation must also be changed according to this.

File: resources/examples/simple_cluster/playbooks/management1.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
 - name: management1
-  hosts: "{{ target | default('management1') }}"
+  hosts: "management1"
```
Comment (bouriquet) [inline resources/examples/simple_cluster/playbooks/management1.yml:3]: same

File: resources/examples/simple_cluster/playbooks/login1.yml (hunk 4 above)
Comment (oxedions) [reply resources/examples/simple_cluster/playbooks/login1.yml:3]: Because I find it dangerous to use target. When adding "target", I didn't knew --limit argument.

I prefer to force a static host or group of hosts in the playbook. Then use --limit to reduce scope.

In production, having a flexible playbook prove to be dangerous, I have seen that just 3 days ago on a cluster I am working on :-)

Yes, Documentation need to be updated, I will do that in a next commit.

Comment (actatux) [reply resources/examples/simple_cluster/playbooks/login1.yml:3]: Maybe use a group (`mg_logins`) instead of a host (`login1`) in the example?

Comment (bouriquet) [reply resources/examples/simple_cluster/playbooks/login1.yml:3]: @actatux yes, that was the point of my suggestion, use login group in hosts field and restrict to login1 using limit option,  i guess this playbook isnt dedicated only to the first login and can be applied on all login nodes, right ?

Comment (actatux) [reply resources/examples/simple_cluster/playbooks/login1.yml:3]: Agree. We might need to rename it `logins.yml`.

Comment (oxedions) [reply resources/examples/simple_cluster/playbooks/login1.yml:3]: Oh, sorry @bouriquet 

I miss read your comment. Thought you were referring to the removal of target.

Then we are all in line :-)

**Comment by oxedions** [comment]
Brainstorming:

```
j2_node_main_network: "{% set main_ntw = [] %}{% for nic in network_interfaces %}{% if network_interfaces[nic].network is defined and not none %}{{ main_ntw.append(network_interfaces[nic].network|string) }}{% endif %}{% endfor %}{{ main_ntw | select('match',(j2_current_iceberg_network+'-[a-zA-Z0-9]+')) | list | unique | sort| first | join | trim }}"

j2_node_main_network_interface: "{% set main_ntw_nic = [] %}{% for nic in network_interfaces %}{% if network_interfaces[nic].network is defined and not none %}{% if network_interfaces[nic].network == j2_node_main_network %}{{ main_ntw_nic.append(nic|string) }}{% endif %}{% endif %}{% endfor %}{{main_ntw_nic | unique | join | trim}}"
```

Replaced by:

```
j2_node_main_network: "{{lookup('dict',network_interfaces) | map(attribute='value') | map(attribute='network') | sort | first }}"

j2_node_main_network_interface: ???
```

**Comment by oxedions** [comment]
Stall. Waiting for another development.

File: resources/examples/multi_icebergs_cluster/inventory/cluster/groups/slave_managements [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Slaves managements, to mount /etc/bluebanquise and repositories from top iceberg
```
Comment (actatux) [inline resources/examples/multi_icebergs_cluster/inventory/cluster/groups/slave_managements:1]: I would rephrase it secondary managements instead of slave.

**Comment by johnnykeats** [comment]
Now taking care of this PR.

@actatux. I will rephrase this part indeed. "Secondary level iceberg's managements" would be perfect.

**Comment by oxedions** [comment]
Thanks a lot @johnnykeats

**Comment by oxedions** [comment]
Can you also cover the update of nic, by adding '- interface:' as in #257 in the multi iceberg example ?
Since files were updated here, to avoid issues when merging.

File: resources/examples/simple_cluster/README [hunk 7]
```diff
@@ -1,7 +1,9 @@
 This examples is composed of:
 
-1 management node
-1 login node
-4 computes nodes
+* 1 management node
+* 1 login node
+* 4 computes nodes
 
 All of them are link with a single ethernet management network, and with an interconnect.
```
Comment (bouriquet) [inline resources/examples/simple_cluster/README:7]: Can you add here the IP address plan of the networks ?

File: resources/examples/multi_icebergs_cluster/README [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+iceberg3 with:
+
+* 1 management node
+* 2 compute nodes on ice3-1 network
+* 2 gpu compute nodes on ice3-2 network
+
+Each iceberg as an isolated ethernet network. Each slave iceberg management have
+on nic on its iceberg network, and one nic on its top iceberg network.
+
```
Comment (bouriquet) [inline resources/examples/multi_icebergs_cluster/README:25]: Can you add here the IP address plan of the networks ?

File: resources/documentation/deploy_bluebanquise.rst [hunk 9]
```diff
@@ -193,8 +229,14 @@ And execute them, using extra var target to target them:
   ansible-playbook /etc/bluebanquise/login1.yml
   ansible-playbook /etc/bluebanquise/computes.yml --extra-vars "target=c001,c002,c003,c004"
```
Comment (bouriquet) [inline resources/documentation/deploy_bluebanquise.rst:230]: "target" var must be removed or replaced by -l/--limit option

**Comment by oxedions** [comment]
JK is offline for few days, I am taking over.

File: resources/examples/simple_cluster/inventory/group_vars/all/general_settings/external.yml [hunk 10]
```diff
@@ -17,13 +17,3 @@ external_dns:
...
   sphenisc.com: 213.186.33.3
-
-# Uncomment this variable if you need additionnal repositories on your clients
-# Be aware that, at this level, theses repositories will be exported on all hosts without
-# any architecture or distribution check.
-
-#external_repositories:
-#  - name: epel
-#    baseurl: 'https://dl.fedoraproject.org/pub/epel/$releasever/$basearch/'
-#    proxy: 'https://proxy:8080'
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/examples/simple_cluster/inventory/group_vars/all/general_settings/external.yml:28]: Yes, let's deprecate them. :-)

File: resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml [hunk 11]
```diff
@@ -3,44 +3,23 @@ nfs_settings:
...
+    mount_arguments: ro,intr,nfsvers=4.2,bg     # Arguments for the client (mount)
 
   repositories:
     mount: /var/www/html/repositories
     export: /var/www/html/repositories
-    server: odin
+    server: management1
     clients_groups:
-      - equipment_supermicro_management
+      - slave_managements
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/examples/multi_icebergs_cluster/inventory/group_vars/all/general_settings/nfs.yml:22]: Done, also corrected the group file related.

File: resources/documentation/install_first_management.rst [hunk 12]
```diff
@@ -150,9 +156,11 @@ BlueBanquise
 
 Download BlueBanquise rpms from official repository.
 
-Go to https://bluebanquise.com, go to repositories/download, and get the content of the whole directory corresponding to your distribution and architecture.
+Go to https://bluebanquise.com, go to repositories/download, and get the content
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/install_first_management.rst:159]: Seems a very good idea to me. Cleaner than using wget.

File: resources/examples/multi_icebergs_cluster/inventory/cluster/groups/slave_managements [hunk 13]
```diff
@@ -0,0 +1,5 @@
+# Secondary level iceberg's managements, to mount /etc/bluebanquise and
+# repositories from top iceberg
+[slave_managements]
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/examples/multi_icebergs_cluster/inventory/cluster/groups/slave_managements:3]: Done

File: resources/documentation/introduction.rst [hunk 14]
```diff
@@ -2,14 +2,22 @@
...
-Ansible was chosen for its simplicity and its security. The Ansible inventory groups/variables mechanism can cover from very simple to very sophisticated configurations. Ansible is not the fastest tool, nor the simplest to debug. However, it is easy to learn, and widely used today with an active community.
+Ansible was chosen for its simplicity and its security.
+The Ansible inventory groups/variables mechanism can cover from very simple to
+very sophisticated configurations. Ansible is not the fastest tool, nor the
+simplest to debug. However, it is easy to learn, and widely used today with an
+active community.
 
-This documentation is structured as a training. If you encounter any bugs/issues or have any comments, please inform us.
+This documentation is structured as a training. If you encounter any bugs/issues
+or have any comments, please inform us.
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/introduction.rst:21]: Done

File: resources/documentation/vocabulary.rst [hunk 15]
```diff
@@ -150,44 +167,59 @@ Jinja2 will be discussed later, do not worry about this point for now.
...
-You can consider these variables as "functions" that takes as argument the current running host (or the host provided in hostvars if loaded using hostvars call).
+These variables are here to simplify tasks and templates writing, by removing
+ugly and redundant things from them, and providing a direct access to values.
+You can consider these variables as "functions" that takes as argument the
+current running host (or the host provided in hostvars if loaded using hostvars
+call).
 
-To clarify your mind, you can consider that these variables contains a simple value. In reality, they contain Jinja2 code as a string, that will be interpreted by Ansible during tasks/templates execution, which is why these are more functions/API than pure variables.
+To clarify your mind, you can consider that these variables contains a simple
+value. In reality, they contain Jinja2 code as a string, that will be
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/vocabulary.rst:183]: I rewamped the whole part. Can you confirm it is ok on your side now ?

[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/vocabulary.rst:183]: (just need to commit, few minutes)

[the remark this answers is not in the document]
Comment (actatux) [reply resources/documentation/vocabulary.rst:183]: wait...what, where is the code ?? ;-)

File: resources/documentation/vocabulary.rst [hunk 16]
```diff
@@ -198,17 +230,31 @@ Variables precedence
...
+loaded.
 
-The full list of available variables precedence is provided in Ansible documentation: `variable precedence list <https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable>`_
+The full list of available variables precedence is provided in Ansible
+documentation:
+`variable precedence list <https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable>`_
 
-This feature is key to the stack and key for system administrator to manipulate the **BlueBanquise** stack the way he/she/(it ?) wants, and *force* automatic values if desired.
+This feature is key to the stack and key for system administrator to manipulate
+the **BlueBanquise** stack the way he/she/(it ?) wants, and *force* automatic
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/vocabulary.rst:251]: Was a jock for AI, but yes, will remove it.

File: resources/examples/simple_cluster/README (hunk 7 above)
Comment (oxedions) [reply resources/examples/simple_cluster/README:9]: I added a simple plan 😊

File: resources/examples/multi_icebergs_cluster/README (hunk 8 above)
Comment (oxedions) [reply resources/examples/multi_icebergs_cluster/README:31]: Same

File: resources/documentation/deploy_bluebanquise.rst (hunk 9 above)
Comment (oxedions) [reply resources/documentation/deploy_bluebanquise.rst]: Ok, corrected.

File: resources/documentation/configure_bluebanquise.rst [hunk 17]
```diff
@@ -224,24 +257,31 @@ General configuration is made in group_vars/all/general_settings.
...
 Repositories
 ^^^^^^^^^^^^
 
-File group_vars/all/general_settings/repositories.yml configure repositories to use for all nodes (using groups and variable precedence, repositories can be tuned for each group of nodes, or even each node).
+File group_vars/all/general_settings/repositories.yml configure repositories to
+use for all nodes (using groups and variable precedence, repositories can be
+tuned for each group of nodes, or even each node).
 
-Right now, only *os* and *bluebanquise* are set. This means two repositories will be added to nodes, and they will bind to repository_ip in ice1-1.yml .
+Right now, only *os* and *bluebanquise* are set. This means two repositories
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/configure_bluebanquise.rst]: Done

File: resources/documentation/deploy_bluebanquise.rst [hunk 18]
```diff
@@ -2,17 +2,21 @@
 Deploy BlueBanquise
 ===================
 
-At this point, **BlueBanquise** configuration is done. We are ready to deploy the cluster.
+At this point, **BlueBanquise** configuration is done. We are ready to deploy
+the cluster.
 
-First step is to deploy configuration on management1 node, and then deploy OS on the others system. Last step will be to deploy configuration on the other systems.
+First step is to deploy configuration on management1 node, and then deploy OS on
+the others system. Last step will be to deploy configuration on the other
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/deploy_bluebanquise.rst]: Done

File: resources/documentation/deploy_bluebanquise.rst [hunk 19]
```diff
@@ -2,17 +2,21 @@
...
 
 Management configuration
 ========================
 
 Get management1 playbook
 ------------------------
 
-We are going to use the provided default playbook. This playbook will install most of the CORE roles. Enough to deploy first stage of the cluster.
+We are going to use the provided default playbook. This playbook will install
+most of the CORE roles. Enough to deploy first stage of the cluster.
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/deploy_bluebanquise.rst]: Done

File: resources/documentation/deploy_bluebanquise.rst [hunk 20]
```diff
@@ -21,7 +25,8 @@ Copy example playbook management1 to /etc/bluebanquise/playbooks/:
   mkdir /etc/bluebanquise/playbooks/
   cp -a /etc/bluebanquise/resources/examples/simple_cluster/playbooks/management1.yml /etc/bluebanquise/playbooks/
 
-Then, we will ask Ansible to read this playbook, and execute all roles listed inside on management1 node (check hosts target at top of the file).
+Then, we will ask Ansible to read this playbook, and execute all roles listed
+inside on management1 node (check hosts target at top of the file).
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/deploy_bluebanquise.rst]: Done

File: resources/documentation/deploy_bluebanquise.rst [hunk 21]
```diff
@@ -35,19 +40,25 @@ We are going to use 2 parameters frequently:
 Tags / Skip tags
 ^^^^^^^^^^^^^^^^
 
-As you can notice, some tags are set inside the playbook, or even in some roles for specific tasks. The idea of tags is simple: you can tag a role/a task, and then when using ansible-playbook, only play related tags role/task. Or do the opposit: play all, and skip a role/task.
+As you can notice, some tags are set inside the playbook, or even in some roles
+for specific tasks. The idea of tags is simple: you can tag a role/a task, and
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/deploy_bluebanquise.rst]: Done

File: resources/documentation/deploy_bluebanquise.rst [hunk 22]
```diff
@@ -35,19 +40,25 @@ We are going to use 2 parameters frequently:
...
 
 To so, use with Ansible playbook:
 
 * **--tags** with tags listed with comma separator: mytag1,mytag2,mytag3
 * **--skip-tags** with same pattern
 
-More can be found `here <https://docs.ansible.com/ansible/latest/user_guide/playbooks_tags.html>`_ on tags.
+More can be found
+`here <https://docs.ansible.com/ansible/latest/user_guide/playbooks_tags.html>`_
+on tags.
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/deploy_bluebanquise.rst]: Done

File: resources/documentation/configure_bluebanquise.rst [hunk 23]
```diff
@@ -252,28 +292,36 @@ File group_vars/all/general_settings/general.yml configure few main parameters:
...
 
-Remember Ansible precedence mechanism. All variables in group_vars/all/ have less priority, while variables in group_vars/* have a higher priority.
+Remember Ansible precedence mechanism. All variables in group_vars/all/ have
+less priority, while variables in group_vars/* have a higher priority.
 
-The idea here is the following: group_vars/all/all_equipments/ folder contains all the default parameters for all nodes. Here authentication, and equipment_profile. You have to tune these parameters to match your exact "global" need, and then tune dedicated parameters for each equipment group.
+The idea here is the following: group_vars/all/equipment_all/ folder contains
+all the default parameters for all nodes. Here authentication, and
+equipment_profile. You have to tune these parameters to match your exact
+"global" need, and then tune dedicated parameters for each equipment group.
```
[the remark this answers is not in the document]
Comment (oxedions) [reply resources/documentation/configure_bluebanquise.rst:311]: Agree. I am correcting that.

**Comment by oxedions** [comment]
Dear @bouriquet , Dear @actatux 

Thanks a lot for these review. I know it takes time.

File: resources/documentation/install_first_management.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (actatux) [reply resources/documentation/install_first_management.rst:159]: #267

File: resources/examples/simple_cluster/inventory/group_vars/all/general_settings/external.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (actatux) [reply resources/examples/simple_cluster/inventory/group_vars/all/general_settings/external.yml:28]: #268

**Comment by oxedions** [comment]
Indeed 😊

File: resources/documentation/deploy_bluebanquise.rst [hunk 24]
```diff
@@ -186,15 +221,21 @@ yes, copy example playbooks:
   cp -a /etc/bluebanquise/resources/examples/simple_cluster/playbooks/computes.yml /etc/bluebanquise/playbooks/
   cp -a /etc/bluebanquise/resources/examples/simple_cluster/playbooks/login1.yml /etc/bluebanquise/playbooks/
```
Comment (bouriquet) [inline resources/documentation/deploy_bluebanquise.rst:222]:
```suggestion
  cp -a /etc/bluebanquise/resources/examples/simple_cluster/playbooks/logins.yml /etc/bluebanquise/playbooks/
```

File: resources/documentation/deploy_bluebanquise.rst [hunk 25]
```diff
@@ -186,15 +221,21 @@ yes, copy example playbooks:
...
 
-And execute them, using extra var target to target them:
+And execute them, using --limit parameter to specify targets them:
 
 .. code-block:: bash
 
   ansible-playbook /etc/bluebanquise/login1.yml
```
Comment (bouriquet) [inline resources/documentation/deploy_bluebanquise.rst:228]:
```suggestion
  ansible-playbook /etc/bluebanquise/logins.yml
```

File: resources/documentation/deploy_bluebanquise.rst [hunk 26]
```diff
@@ -21,7 +25,8 @@ Copy example playbook management1 to /etc/bluebanquise/playbooks/:
   mkdir /etc/bluebanquise/playbooks/
   cp -a /etc/bluebanquise/resources/examples/simple_cluster/playbooks/management1.yml /etc/bluebanquise/playbooks/
```
Comment (bouriquet) [inline resources/documentation/deploy_bluebanquise.rst:26]:
```suggestion
  cp -a /etc/bluebanquise/resources/examples/simple_cluster/playbooks/managements.yml /etc/bluebanquise/playbooks/
```

File: resources/documentation/deploy_bluebanquise.rst [hunk 27]
```diff
@@ -2,17 +2,21 @@
...
 ========================
 
 Get management1 playbook
 ------------------------
 
-We are going to use the provided default playbook. This playbook will install most of the CORE roles. Enough to deploy first stage of the cluster.
+We are going to use the provided default playbook. This playbook will install
+most of the **core** roles. Enough to deploy first stage of the cluster.
 
 Copy example playbook management1 to /etc/bluebanquise/playbooks/:
```
Comment (bouriquet) [inline resources/documentation/deploy_bluebanquise.rst:21]:
```suggestion
Copy example playbook managements to /etc/bluebanquise/playbooks/:
```

File: resources/documentation/deploy_bluebanquise.rst [hunk 28]
```diff
@@ -56,7 +66,8 @@ To do so, use:
 Apply management1 configuration
 -------------------------------
 
-Lets apply now the whole configuration on management1. It can takes some time depending of your CPU and your hard drive.
+Lets apply now the whole configuration on management1. It can take some time
+depending on your CPU and your hard drive.
 
 We first ensure our NIC are up, so the repositories part is working.
```
Comment (bouriquet) [inline resources/documentation/deploy_bluebanquise.rst:73]: Replace 

ansible-playbook /etc/bluebanquise/playbooks/management1.yml --tags set_hostname,nic

by 

ansible-playbook /etc/bluebanquise/playbooks/managements.yml --tags set_hostname,nic

File: resources/documentation/deploy_bluebanquise.rst [hunk 29]
```diff
@@ -70,7 +81,8 @@ Then start your main interface manually. Here enp0s3:
 
   ifup enp0s3
 
-Once interface is up (check using *ip a* command), setup the BlueBanquise controller:
+Once interface is up (check using *ip a* command), setup the BlueBanquise
+controller:
 
 .. code-block:: bash
```
Comment (bouriquet) [inline resources/documentation/deploy_bluebanquise.rst:88]: Replace 

ansible-playbook /etc/bluebanquise/playbooks/management1.yml --tags bluebanquise

by 

ansible-playbook /etc/bluebanquise/playbooks/managements.yml --tags bluebanquise

Comment (bouriquet) [reply resources/documentation/deploy_bluebanquise.rst:88]: Replace (L95) :

ansible-playbook /etc/bluebanquise/playbooks/management1.yml

by 

ansible-playbook /etc/bluebanquise/playbooks/managements.yml

File: resources/documentation/deploy_bluebanquise.rst (hunk 28 above)
Comment (actatux) [reply resources/documentation/deploy_bluebanquise.rst:73]: Done, with `--limit management1`
